react/__tests__/components/BuyButton.test.js

```diff
@@ -34,4 +35,28 @@ describe('<BuyButton />', () => {
     const { asFragment } = renderComponent({ available: false })
     expect(asFragment()).toMatchSnapshot()
   })
+
+  it('should call onAddStart and onAddFinish', done => {
+    const onAddStart = jest.fn()
+    const onAddFinish = jest.fn()
+
+    const buttonText = 'Test inside button'
+
+    const { getByText } = renderComponent(
+      {
+        available: true,
+        skuItems: [],
+        onAddStart,
+        onAddFinish,
+      },
+      buttonText
+    )
+
+    fireEvent.click(getByText(buttonText))
+    setTimeout(() => {
```

Comment:
Why are you using setTimeout?

Reply:
Because this `onClick` is asynchronous and don't modify the DOM so I can't use `wait` or `waitForElement` if you have any suggestion, I will accept.